kmelia/kmelia-war/src/main/java/org/silverpeas/components/kmelia/control/KmeliaSessionController.java

```diff
@@ -3736,4 +3752,128 @@ public void updatePublicationReminder(String pubId, List<FileItem> parameters)
       createResourceDateReminder(pubDetail, dateReminderDate, messageReminder);
     }
   }
+
+  public boolean isTemplatesSelectionEnabledForRole(SilverpeasRole role) {
+    return !isXMLFormEnabledForPublications() && isContentEnabled() &&
+        SilverpeasRole.admin.equals(role);
+  }
+
+  private String getXMLFormNameForPublications() {
+    return getComponentParameterValue("XmlFormForPublis");
+  }
+
+  private boolean isXMLFormEnabledForPublications() {
+    return StringUtil.isDefined(getXMLFormNameForPublications());
+  }
+
+  public Form getXmlFormForPublications() {
+    Form formUpdate = null;
+    List<String> forms = getModelUsed();
+    if (forms.size() == 1 && !forms.get(0).equals("WYSIWYG")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the "WYSIWYG" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)